### PR TITLE
Retry failed dismounts with new destination slots

### PR DESCRIPTION
**Pull request by renbauer-fnal**
locate_volume now takes a new argument which allows callers to exclude slots from result set. This allows mtx_dismount to retry while excluding previously failed addresses from its search for EMPTY slots.

Combined with the addition of `Media magazine not accessible` as a retriable error, this resolves failures from attempting to dismount into inaccessible slots.

### CODE REVIEW

File: src/media_changer.py [hunk 1]
```diff
@@ -5929,7 +5938,7 @@ def mtx_dismount(self, volume, drive, media_type="", view_first=1):
     # numbers are not negative one then there is either a bug in this
     # function or multiple tapes that have the same label in the
     # library.
-    def locate_volume(self, vol):
+    def locate_volume(self, vol, ignore_addresses=[]):
         Trace.trace(ACTION_LOG_LEVEL, ' looking for volume %s status_valid %s' % (vol, self.status_valid))
```
Comment (DmitryLitvintsev) [inline src/media_changer.py:5946]: I think this is python pitfall number one - never use mutable default argument


(https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

replace with : 
```
def locate_volume(self, vol, ignore_addresses=None):
```

Comment (renbauer-fnal) [reply src/media_changer.py:5946]: Ah, thanks - PyCharm has helped me fix enough of these throughout the codebase I should know this by now. Will change now

Comment (renbauer-fnal) [reply src/media_changer.py:5946]: Resolved in [0e7292d](https://github.com/Enstore-org/enstore/pull/118/commits/0e7292dc4adc19dbe5cd37103fb2c74333e162e7)

File: src/media_changer.py [hunk 2]
```diff
@@ -3924,8 +3924,9 @@ def do_retry(self, ret_val, returned_by_mtx_mount=False):
                         update_db = True
                     else:
                         if 'mtx: Request Sense: Additional Sense Code = 3B' in ret_val[3]:
-                            if 'mtx: Request Sense: Additional Sense Qualifier = 0D' in ret_val[3]:
```
Comment (DmitryLitvintsev) [inline src/media_changer.py:3927]: Frankly I do not understand this code 
```
if "a" in ret_val[3]:
   if "b" in  ret_val[3]:
```
what is ret_val[3] ? is that a list like ("a", "b", ...)?
I assume this is , but just wanted to ask is it?

**Review by DmitryLitvintsev — APPROVED** [review]

File: src/media_changer.py (hunk 2 above)
Comment (renbauer-fnal) [reply src/media_changer.py:3927]: Yes, AIUI ret_val is the tuple in this error message:
```
[enstore@-- ~]$ enstore med --dismount VOL 278 LIB
BAD STATUS (1, 1, '', ['', 'mtx: Request Sense: Long Report=yes', 'mtx: Request Sense: Valid Residual=no', 'mtx: Request Sense: Error Code=70 (Current)', 'mtx: Request Sense: Sense Key=Illegal Request', 'mtx: Request Sense: FileMark=no', 'mtx: Request Sense: EOM=no', 'mtx: Request Sense: ILI=no', 'mtx: Request Sense: Additional Sense Code = 3B', 'mtx: Request Sense: Additional Sense Qualifier = 11', 'mtx: Request Sense: BPV=no', 'mtx: Request Sense: Error in CDB=no', 'mtx: Request Sense: SKSV=no', 'MOVE MEDIUM from Element Address 278 to 6920 Failed', ''])
```

So, ret_val[3] is a list of.. error strings returned by SCSI, as parsed by mtx, I think.

Comment (renbauer-fnal) [reply src/media_changer.py:3927]: I've tried to change the error handling implementation as little as possible

**Comment by DmitryLitvintsev** [comment]
I forget. Our policy that you squash & merge. I approved the PR. (just making sure this is not dependent on me)

**Pull request merged**